crates/sui-graphql-rpc/schema.graphql

```diff
@@ -0,0 +1,840 @@
+type Object implements Owner {
+  owner: AmbiguousOwner
+  version: U64
+  digest: Digest
+
+  previousVersion: U64
+  previousTransactionBlock: TransactionBlock
+  storageRebate: U64
+
+  display: [DisplayEntry]
+
+  asMoveObject: MoveObject
+  asMovePackage: MovePackage
+
+  # Transaction Blocks that sent objects to this object
+  receivedTransactionBlocks: TransactionBlockConnection
+
+  dynamicField(dynamicFieldName: DynamicFieldName!): DynamicField
+  dynamicFields: DynamicFieldConnection
+
+  bcs: Bytes
+}
+
+type DisplayEntry {
+  key: String
+  value: String
+}
+
+type Epoch {
+  epochId: Int!
+  systemStateVersion: Int
+  protocolConfig: ProtocolConfig
+  referenceGasPrice: U64
+
+  systemParameters: SystemParameters
+  stakeSubsidy: StakeSubsidy
+  validatorSet: ValidatorSet
+  storageFund: StorageFund
+  safeMode: SafeMode
+
+  startTimestampMs: U64
+  endTimestampMs: U64
+
+  # TODO: Identify non-duplicate fields in `EndOfEpochInfo`
+
+  checkpoints: CheckpointConnection
+  transactionBlocks: TransactionBlockConnection
+}
+
+type ProtocolConfig {
+  protocolVersion: Int!
+  # Omitting for brevity
+}
+
+type SystemParameters {
+  durationMs: U64
+  stakeSubsidyStartEpoch: Int
+
+  minValidatorCount: Int
+  maxValidatorCount: Int
+
+  minValidatorJoiningStake: U64
+  validatorLowStakeThreshold: U64
+  validatorVeryLowStakeThreshold: U64
+  validatorLowStakeGracePeriod: Int
+}
+
+type StakeSubsidy {
+  balance: U64
+  distributionCounter: U64
+  currentDistributionAmount: U64
+  periodLength: Int
+  decreaseRate: Int
+}
+
+type ValidatorSet {
+  totalStake: U64
+
+  activeValidators: [Validator]
+
+  # Indices into `activeValidators`
+  pendingRemovals: [U64]
+
+  pendingActiveValidators: MoveObject
+  pendingActiveValidatorsSize: U64
+
+  stakePoolMappings: MoveObject
+  stakePoolMappingsSize: U64
+
+  inactivePools: MoveObject
+  inactivePoolsSize: U64
+
+  validatorCandidates: MoveObject
+  validatorCandidatesSize: U64
+}
+
+type Validator {
+  address: Address!
+
+  credentials: ValidatorCredentials
+  nextEpochCredentials: ValidatorCredentials
+
+  name: String
+  description: String
+  imageUrl: String
+  projectUrl: String
+
+  operationCap: MoveObject
+  stakingPool: MoveObject
+  exchangeRates: MoveObject
+
+  stakingPoolActivationEpoch: Int
+  stakingPoolSuiBalance: U64
+  rewardsPool: U64
+  poolTokenBalance: U64
+  pendingStake: U64
+  pendingTotalSuiWithdraw: U64
+  pendingPoolTokenWithdraw: U64
+  exchangeRatesSize: U64
+
+  votingPower: Int
+  stakeUnits: Int
+  gasPrice: U64
+  commissionRate: U64
+  nextEpochStake: U64
+  nextEpochGasPrice: U64
+  nextEpochCommissionRate: U64
+
+  # The number of epochs for which this validator has been below the
+  # low stake threshold.
+  atRisk: Int
+
+  # The other validators this validator has reported
+  reportRecords: [SuiAddress]
+
+  apy: Float
+}
+
+type ValidatorCredentials {
+  protocolPubKey: Bytes
+  networkPubKey: Bytes
+  workerPubKey: Bytes
+  proofOfPossession: Bytes
```

Comment:
In the current RPC, these are Base64, and this signature implies these become Base58, to avoid us having to introduce a new scalar type just for them.